build/build-image/Dockerfile

```diff
@@ -36,6 +36,13 @@ ENV KUBE_OUTPUT_SUBPATH _output/dockerized
 # Pick up version stuff here as we don't copy our .git over.
 ENV KUBE_GIT_VERSION_FILE ${HOME}/.dockerized-kube-version-defs
 
+# Add system-wide git user information
+RUN git config --system user.email "nobody@k8s.io" \
+  && git config --system user.name "kube-build-image"
+
+# Fix permissions on gopath
+RUN chmod -R a+rwx $GOPATH
```

Comment:
The reason for this is here: https://github.com/kubernetes/kubernetes/pull/46193/files#diff-8188c017a3a4a9343bc9920417e2376bR70

We're `go get`ing these tools as root when the kube-cross container is built, which breaks the perms on the GOPATH when we try to restore godeps